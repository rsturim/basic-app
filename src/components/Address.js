import React from 'react';

import PropTypes from 'prop-types';

function renderItem(type, item) {
    if (!item) {
        return;
    } else {
        return <div type={type}>{item}</div>;
    }
}

export default function Address(props) {
    const { person, city, street, state, zip } = props;

    return (
        <address>
            {renderItem('person', person)}
            {renderItem('street', street)}
            {renderItem('city', city)}
            {renderItem('state', state)}
            {renderItem('zip', zip)}
        </address>
    );
}

Address.propTypes = {
    person: PropTypes.string.isRequired,
    street: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
    state: PropTypes.string.isRequired,
    zip: PropTypes.string.isRequired,
};
