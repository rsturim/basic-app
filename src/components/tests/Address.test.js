import React from 'react';
import { mount } from 'enzyme';

import Address from '../Address';
import toJson from 'enzyme-to-json';

describe('An addres component', () => {
    let wrapper = {};
    beforeEach(() => {
        wrapper = mount(
            <Address
                person="Joe Smith"
                street="123 Cupcake lane"
                city="Emerald City"
                state="VT"
                zip="05445"
            />,
        );
    });

    test('should derive props from a mounted component ', () => {
        expect(wrapper.prop('person')).toEqual('Joe Smith');
        expect(wrapper.props().street).toEqual('123 Cupcake lane');
        expect(wrapper.props().city).toEqual('Emerald City');
        expect(wrapper.props().state).toEqual('VT');
        expect(wrapper.props().zip).toEqual('05445');

        // wrapper.unmount();
    });

    test('should also match the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
