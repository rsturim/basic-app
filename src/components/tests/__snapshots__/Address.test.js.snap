// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`An addres component should also match the snapshot 1`] = `
<Address
  city="Emerald City"
  person="Joe Smith"
  state="VT"
  street="123 Cupcake lane"
  zip="05445"
>
  <address>
    <div
      type="person"
    >
      Joe Smith
    </div>
    <div
      type="street"
    >
      123 Cupcake lane
    </div>
    <div
      type="city"
    >
      Emerald City
    </div>
    <div
      type="state"
    >
      VT
    </div>
    <div
      type="zip"
    >
      05445
    </div>
  </address>
</Address>
`;
